### PR TITLE
Dashboard: handle error when when user has no permission to view site visit stats

### LINKED ISSUE #823: Dashboard: Shop Manager has persistent "Unable to load content" error from site stats
**Describe the bug**

When I’m logged in to the app as a user with the Shop Manager (not admin) role, on the “My store” tab I see the notice “Unable to load content.” It looks like this is related to the store stats not loading, which shows up in the app logs and with the Visitors data not loading in the stats chart.

Errors in the logs:

```
2019/03/28 12:06:24:673  ⛔️ Dashboard (Site Stats) — Error synchronizing site visit stats: Dotcom Missing Token
2019/03/28 12:06:24:674  ⛔️ Error synchronizing visitor stats: Dotcom Missing Token
```

**To Reproduce**
1. Add a user to your store with the [Shop Manager role](https://docs.woocommerce.com/document/roles-capabilities/#section-2).
1. Log in as that user in a web browser and connect to Jetpack.
1. Log in as that user in the app. Note that there are no warnings or anything to prevent you from logging in to the app with this role.
1. Select the store where you have the Shop Manager role.
1. Visit the "My store" tab. Note that Visitors doesn't load in the chart and you get an error notice.
1. Navigate around the app and confirm that other features work as expected, but the error notice recurs each time you open the My store tab again.

**Screenshots**

image

**Expected behavior**

I expect the app to exclude features that I don't have permissions to use when I log in as a Shop Manager.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone XS
- iOS version: 12.1.4
- WooCommerce iOS version: 1.5.0.0 (TestFlight beta)

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #823 

## Changes
- In `DotcomError`, added the following two cases. Unfortunately, our dotcom [API error identifier is not unique](https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/) for each error scenario so there is code to check the error message
  - `noStatsPermission`: when the user has no permission to view site stats
  - `statsModuleDisabled`: when Jetpack site stats module disabled
- Added `noPermission` error case to `SiteVisitStatsStoreError`, and mapped the new `DotcomError` cases (in the Networking layer) to the corresponding `SiteVisitStatsStoreError` cases (in the Yosemite layer)
- Handled `noPermission` site visit stats error case the same as when stats module is disabled (https://github.com/woocommerce/woocommerce-ios/pull/1110 https://github.com/woocommerce/woocommerce-ios/pull/1105) - hiding visitor stats UI without alert toast
- Updated unit tests

## Testing
- Prerequisite: have a store with Jetpack installed, with another store user of `Shop Manager` role (could be any other role than `admin`, and the following steps use `Shop Manager` as the example)
- Go to Jetpack modules settings page at your store's `/wp-admin/admin.php?page=jetpack_modules`, and find the "site stats" module. Make sure the site stats module is activated
- Go to your store's `/wp-admin/admin.php?page=jetpack#/traffic`, expand the "site stats" section, and make sure `Shop manager` is turned off under `Allow stats reports to be viewed by` subsection
- Launch the app with the store --> there should not be a generic alert as in the screenshot [here](https://github.com/woocommerce/woocommerce-ios/issues/932#issuecomment-488332137) and the visitor stats should be hidden
- Go back to your store's `/wp-admin/admin.php?page=jetpack#/traffic`, "site stats" section, and turn on `Shop manager` under `Allow stats reports to be viewed by` subsection
- Go back to the app with the store and pull down to refresh on the Dashboard --> visitor stats should be shown again
- You can go to Jetpack modules settings page again at your store's `/wp-admin/admin.php?page=jetpack_modules`, and deactivate "site stats" module. Going back to the app, the visitor stats should be hidden again (for the reason of stats module disabled) without alert toast

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by jaclync** [comment]
Ready for review now @ctarda @mindgraffiti @jleandroperez ! feedback appreciated 😄

File: Networking/Networking/Model/DotcomError.swift [hunk 1]
```diff
@@ -29,34 +29,53 @@ public enum DotcomError: Error, Decodable {
...
             self = .invalidToken
         case Constants.requestFailed:
             self = .requestFailed
         case Constants.unauthorized:
-            self = .unauthorized
+            if message == ErrorMessages.noStatsPermission {
+                self = .noStatsPermission
+            } else {
+                self = .unauthorized
+            }
```
Comment (jleandroperez) [inline Networking/Networking/Model/DotcomError.swift]: Nipticky tip!

```
        case Constants.unauthorized where message == ErrorMessages.noStatsPermission:
            self = .noStatsPermission
        case Constants.unauthorized:
            self = .unauthorized
```

(That way you can get rid of the if!)

File: Networking/Networking/Model/DotcomError.swift [hunk 2]
```diff
@@ -29,34 +29,53 @@ public enum DotcomError: Error, Decodable {
...
         case Constants.noRestRoute:
             self = .noRestRoute
+        case Constants.invalidBlog:
+            if message == ErrorMessages.statsModuleDisabled {
+                self = .statsModuleDisabled
+            } else {
+                self = .unknown(code: error, message: message)
+            }
```
Comment (jleandroperez) [inline Networking/Networking/Model/DotcomError.swift]: Same as above:

```
        case Constants.invalidBlog where message == ErrorMessages.statsModuleDisabled:
            self = .statsModuleDisabled
        case Constants.invalidBlog:
            self = .unknown(code: error, message: message)
```

Isn't Swift beautiful?

**Review by ctarda — APPROVED** [review]
Sorry for getting so late to this 😊 

Works 💯  and the code looks 💯  as well!

:shipit:

File: Networking/Networking/Model/DotcomError.swift (hunk 2 above)
Comment (jaclync) [reply Networking/Networking/Model/DotcomError.swift]: so much better looking!

**Pull request merged**